cpp/src/gandiva/convert_timezone_holder.h

```diff
@@ -0,0 +1,282 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "arrow/vendored/datetime/date.h"
+#include "arrow/vendored/datetime/tz.h"
+#include "gandiva/function_holder.h"
+#include "gandiva/node.h"
+#include "gandiva/visibility.h"
+
+using arrow_vendored::date::local_time;
+using arrow_vendored::date::locate_zone;
+using arrow_vendored::date::make_zoned;
+using arrow_vendored::date::sys_time;
+using arrow_vendored::date::time_zone;
+using std::chrono::milliseconds;
+
+namespace gandiva {
+
+class OffsetZone {
+  std::chrono::seconds offset_;
+
+ public:
+  explicit OffsetZone(std::chrono::seconds offset) : offset_(offset) {}
+
+  template <class Duration>
+  auto to_local(arrow_vendored::date::sys_time<Duration> tp) const {
+    using std::common_type_t;
+    using std::chrono::seconds;
+    using LT = local_time<std::common_type_t<Duration, seconds>>;
```

Comment:
You add `using std::common_type_t` in the first line, but you repeat the std:: again. Remove the first line or remove the std:: in the last one

Reply:
done